src/ota/ota_handler.rs

```diff
@@ -233,23 +233,54 @@ impl<'a> OTAHandler<'a> {
         if self.state_repository.exists() {
             info!("Found pending update");
             let state = self.state_repository.read()?;
+            let result_response = match self.do_pending_ota(&state).await {
+                Ok(()) => {
+                    info!("OTA successful");
+                    self.send_ota_response(sdk, &state.uuid, OTAStatus::Done)
+                        .await
+                }
+                Err(error) => {
+                    warn!("OTA failed, error -> {:?}", error);
+                    self.send_ota_response(sdk, &state.uuid, OTAStatus::Error(OTAError::Failed))
+                        .await
+                }
+            };
 
-            if state.slot != self.ota.boot_slot().await? {
-                info!("OTA successful");
-                self.send_ota_response(sdk, &state.uuid, OTAStatus::Done)
-                    .await?;
-            } else {
-                warn!("OTA failed");
-                self.send_ota_response(sdk, &state.uuid, OTAStatus::Error(OTAError::Failed))
-                    .await?;
+            if result_response.is_err() {
+                warn!(
+                    "Unable to publish OTA response -> {:?}",
+                    result_response.err()
+                );
```

Comment:
Should this be retried (either today or in the future, leaving a TODO) on the device side or should the cloud side handle a missing answer from the device with a timeout? cc @bettio

Reply:
Let's leave a TODO, but at least I would love avoiding timeouts.